api/tests/unit/mappers/test_unit_mappers_engine.py

```diff
@@ -0,0 +1,463 @@
+from datetime import datetime
+from typing import TYPE_CHECKING
+
+import pytest
+from flag_engine.environments.integrations.models import IntegrationModel
+from flag_engine.environments.models import (
+    EnvironmentAPIKeyModel,
+    EnvironmentModel,
+    WebhookModel,
+)
+from flag_engine.features.models import (
+    FeatureModel,
+    FeatureSegmentModel,
+    FeatureStateModel,
+    MultivariateFeatureOptionModel,
+    MultivariateFeatureStateValueModel,
+)
+from flag_engine.identities.models import IdentityModel, TraitModel
+from flag_engine.organisations.models import OrganisationModel
+from flag_engine.projects.models import ProjectModel
+from flag_engine.segments.models import (
+    SegmentConditionModel,
+    SegmentModel,
+    SegmentRuleModel,
+)
+from pytest_mock import MockerFixture
+
+from environments.models import Environment
+from features.models import FeatureSegment, FeatureState
+from integrations.common.models import IntegrationsModel
+from integrations.dynatrace.models import DynatraceConfiguration
+from integrations.webhook.models import WebhookConfiguration
+from mappers import engine
+from segments.models import Segment, SegmentRule
+
+if TYPE_CHECKING:  # pragma: no cover
+    from environments.identities import Identity, Trait
+    from environments.models import EnvironmentAPIKey
+    from features.models import Feature
+    from projects.models import Project
+
+
+@pytest.fixture()
+def segment_multivariate_feature_state(
+    project: "Project",
+    environment: Environment,
+    multivariate_feature: "Feature",
+) -> FeatureState:
+    segment = Segment.objects.create(
+        name="segment",
+        project=project,
+        feature=multivariate_feature,
+    )
+    FeatureSegment.objects.create(
+        feature=multivariate_feature,
+        segment=segment,
+        environment=environment,
+    )
+    return multivariate_feature.feature_states.first()
```

Comment:
Unless I'm mistaken, I don't think this fixture is returning what you're expecting it to. There are 2 reasons why this is the case: 

 1. A `FeatureSegment` model instance on it's own does nothing, you need to create a `FeatureState` linked to the `FeatureSegment`
 2. `multivariate_feature.feature_states.first()` will always return the environment default, since the default ordering is by id and the env default feature state is always created first.

Reply:
Fixed.